python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_xgboost.py

```diff
@@ -127,6 +124,13 @@ def test_predict(device: str) -> np.ndarray:
     predt0 = reg.predict(X_df)
 
     predt1 = booster.inplace_predict(X_df)
+    # After https://github.com/dmlc/xgboost/pull/11014, .inplace_predict()
+    # returns a real cupy array when called on a cudf.pandas proxy dataframe.
+    # So we need to ensure we have a valid numpy array.
+    # TODO: We should remove the call to .get() when .inplace_predict()
```

Comment:
Is this TODO something we need to change in cudf.pandas, or is it a bug in xgboost?

Reply:
We'd need to make a change in xgboost. Let me cc @galipremsagar to get his thoughts. I believe `.inplace_predict()` should return a cudf.pandas proxy ndarray.

Reply:
Related issue: https://github.com/rapidsai/cudf/issues/17524

Reply:
I don't think we can make `.inplace_predict()` return a `cudf.pandas` proxy object. Xgboost takes different code-paths for `cudf` & `pandas` objects, for a library that executes object(gpu/cpu) aware, I don't think there is much we can do rather than the current state of fixes present in xgboost. But I also feel `predict` has very old code and it needs to be updated to match `inplace_predict` behavior. So I opened a pr asking for reviews: https://github.com/dmlc/xgboost/pull/11118